torch/csrc/inductor/aot_inductor_tensor.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <torch/csrc/inductor/aot_inductor_utils.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef enum {
+  kAotInductorCPU = 0,
+  kAotInductorCUDA = 1,
+} AotInductorDeviceType;
+
+typedef enum {
+  kAotInductorByte = 0,
+  kAotInductorChar = 1,
+  kAotInductorShort = 2,
+  kAotInductorInt = 3,
+  kAotInductorLong = 4,
+  kAotInductorHalf = 5,
+  kAotInductorFloat = 6,
+  kAotInductorDouble = 7,
+  kAotInductorComplexHalf = 8,
+  kAotInductorComplexFloat = 9,
+  kAotInductorComplexDouble = 10,
+  kAotInductorBool = 11,
+  kAotInductorQInt8 = 12,
+  kAotInductorQUInt8 = 13,
+  kAotInductorQInt32 = 14,
+  kAotInductorBFloat16 = 15,
+} AotInductorScalarType;
+
+typedef struct {
+  AotInductorDeviceType device_type;
+  int8_t device_id;
+} AotInductorDevice;
+
+// Bare minumum tensor struct to interchange data between AOTInducotor
+// generated shared library and libtorch to avoid ABI compatibility issues
```

Comment:
If you want other people to be able to work on this file, you're going to need a more detailed description about what exactly the ABI guarantees are, and under what circumstances you can change the declarations in this file. This is especially important because (1) we don't have that many C++ experts on the project and (2) we have even fewer people who regularly have to deal with ABI stable code. fbcode peeps are even worse, since they are spoiled by the monorepo setup. The default assumption should be that someone touching this code has no idea about the ABI concerns and you need big flashing warnings on this file (and any OTHER files) about what is and not OK to do.